### PR TITLE
PIP-1312-test-with-caper

### PR DESCRIPTION
**Pull request by ottojolanki**
Move testing to use caper instead of old script that was running cromwell directly. After this all the new tests can be written in pytest workflow.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -35,6 +35,54 @@ get_kallisto_index: &get_kallisto_index
...
+      - run: pip install -r test_env_requirements.txt
+
+  download_file:
+    description: "Download a file using wget"
+    parameters:
+      file_url:
+        type: string
+    steps:
+      - run: wget <<parameters.file_url>>
+
```
Comment (paul-sud) [inline .circleci/config.yml:63]: Could get rid of this command by passing JAR URLs directly to caper

File: .circleci/config.yml [hunk 2]
```diff
@@ -35,6 +35,54 @@ get_kallisto_index: &get_kallisto_index
...
+  prepare_pytest_environment:
+    description: "Get machine ready to run caper and pytest-workflow"
+    steps:
+      - set_env_variables
+      - install_python_requirements
+      - download_file:
+          file_url: "https://github.com/broadinstitute/cromwell/releases/download/56/cromwell-56.jar"
+      - download_file:
+          file_url: "https://github.com/broadinstitute/cromwell/releases/download/56/womtool-56.jar"
+
```
Comment (paul-sud) [inline .circleci/config.yml:73]: Could avoid downloading jars by passing the URLs directly to caper

File: test/caper_run.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+set -e
+
+echo "Running cromwell $CROMWELL, womtool $WOMTOOL and image $TAG"
+
+if [[ $# -eq 1 ]]; then
+    caper run $1 --docker $TAG --cromwell $CROMWELL --womtool $WOMTOOL --options ./test/cromwell_options.json -m metadata.json
+fi
+
+if [[ $# -eq 2 ]]; then
+    caper run $1 --docker $TAG --cromwell $CROMWELL --womtool $WOMTOOL -i $2 --options ./test/cromwell_options.json -m metadata.json
```
Comment (paul-sud) [inline test/caper_run.sh:12]: Same as above.

**Review by paul-sud — CHANGES_REQUESTED** [review]
Most of the comments here are related to how the JARs are set up for caper. It seems like it would be a simpler to pass caper the JAR URLs directly (see comment on test/caper_run.sh). Other than that just some small things.

File: test/caper_run.sh (hunk 3 above)
Comment (ottojolanki) [reply test/caper_run.sh:12]: But if I don't want to download these every time I run them locally?

File: test_env_requirements.txt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+caper==0.8.2.1
+pytest==5.4.3
+pytest-workflow @ git+https://github.com/LUMC/pytest-workflow.git@4cbc3bb6b9840dafdfc584d62c1e7f9503f6291b
```
[the remark this answers is not in the document]
Comment (ottojolanki) [reply test_env_requirements.txt:3]: yeah, that's it. gonna change.

File: test/caper_run.sh (hunk 3 above)
Comment (paul-sud) [reply test/caper_run.sh:12]: Caper will cache them in `~/.caper/{cromwell_jar,local_jar}`

File: test/caper_run.sh [hunk 5]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+set -e
+
+echo "Running cromwell $CROMWELL, womtool $WOMTOOL and image $TAG"
+
+if [[ $# -eq 1 ]]; then
+    caper run $1 --docker $TAG --cromwell $CROMWELL --womtool $WOMTOOL --options ./test/cromwell_options.json -m metadata.json
```
[the remark this answers is not in the document]
Comment (ottojolanki) [reply test/caper_run.sh:8]: I like having it in the open in the config yaml which versions i am running the tests with. How strongly do you feel about this?

File: .circleci/config.yml [hunk 6]
```diff
@@ -35,6 +35,54 @@ get_kallisto_index: &get_kallisto_index
...
+commands:
+  set_env_variables:
+    description: "Set environment ready for docker running and pushing"
+    steps:
+      - run:
+          command: |
+            echo "export TAG=encodedcc/${CIRCLE_PROJECT_REPONAME}:${CIRCLE_BRANCH}_${CIRCLE_WORKFLOW_ID}" >> ${BASH_ENV}
+            echo "export CROMWELL=./cromwell-56.jar" >> ${BASH_ENV}
+            echo "export WOMTOOL=./womtool-56.jar" >> ${BASH_ENV}
+            source ${BASH_ENV}
```
[the remark this answers is not in the document]
Comment (ottojolanki) [reply .circleci/config.yml]: That is true, thanks.

File: test/caper_run.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (paul-sud) [reply test/caper_run.sh:8]: Discussed in slack, it will be left as is.

**Review by paul-sud — APPROVED** [review]
LGTM

**Pull request merged**